ecs-cli/modules/utils/regcreds/creds_input_reader.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package readers
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// ECSRegCredsInput contains registry cred entries for creation and/or use in a task execution role
+type ECSRegCredsInput struct {
+	Version             string
+	RegistryCredentials RegistryCreds `yaml:"registry_credentials"`
+}
+
+// RegistryCreds is a map of registry names to RegCredEntry structs
+type RegistryCreds map[string]RegistryCredEntry
+
+// RegistryCredEntry contains info needed to create an AWS Secrets Manager secret and match it to an ECS container(s)
+type RegistryCredEntry struct {
```

Comment:
For sake of verbosity, any reason not to call this simply `RegistryCred`?

Reply:
1. It feels too close to `RepositoryCred[ential]` which is a specific field in the task def.
2. Kinda feels like a misnomer - it's not necessarily a singular thing, but a set of information we're going to use make to decisions re: creating/updating/accommodating permissions for a specific registry.